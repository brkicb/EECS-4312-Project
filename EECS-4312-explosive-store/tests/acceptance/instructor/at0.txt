-- A basic example
new_store(6,6)
put("A1",[1,1],[0,0])
put("B1",[2,1],[1,0])
put("C1",[3,1],[3,0])
put("A2",[1,2],[0,1])
put("A3",[1,3],[0,3])
put("A3",[1,3],[0,3])  -- error ID already exists
put("a9",[1,3],[0,3])  -- error explosive will overlap
suggest([1,1])
suggest([2,2])
put("z0", [1,1],[3,4])
suggest([2,2])
put("B2",[2,2],[1,1])
put("B3",[2,3],[1,3])
put("C2",[3,2],[3,1])
put("C3",[3,3],[3,3])
remove("B2")
remove("A1")
remove("C3")
suggest([2,2])
suggest([1,1])
suggest([3,3])
