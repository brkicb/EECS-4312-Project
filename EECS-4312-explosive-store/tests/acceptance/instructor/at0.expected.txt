  System State: (0) Create a new store
->new_store(6,6)
  System State: (1) 
  __ __ __ __ __ __ 
  __ __ __ __ __ __ 
  __ __ __ __ __ __ 
  __ __ __ __ __ __ 
  __ __ __ __ __ __ 
  __ __ __ __ __ __ 
->put("A1",[1, 1],[0, 0])
  System State: (2) 
  __ __ __ __ __ __ 
  __ __ __ __ __ __ 
  __ __ __ __ __ __ 
  __ __ __ __ __ __ 
  __ __ __ __ __ __ 
  A1 __ __ __ __ __ 
->put("B1",[2, 1],[1, 0])
  System State: (3) 
  __ __ __ __ __ __ 
  __ __ __ __ __ __ 
  __ __ __ __ __ __ 
  __ __ __ __ __ __ 
  __ __ __ __ __ __ 
  A1 B1 B1 __ __ __ 
->put("C1",[3, 1],[3, 0])
  System State: (4) 
  __ __ __ __ __ __ 
  __ __ __ __ __ __ 
  __ __ __ __ __ __ 
  __ __ __ __ __ __ 
  __ __ __ __ __ __ 
  A1 B1 B1 C1 C1 C1 
->put("A2",[1, 2],[0, 1])
  System State: (5) 
  __ __ __ __ __ __ 
  __ __ __ __ __ __ 
  __ __ __ __ __ __ 
  A2 __ __ __ __ __ 
  A2 __ __ __ __ __ 
  A1 B1 B1 C1 C1 C1 
->put("A3",[1, 3],[0, 3])
  System State: (6) 
  A3 __ __ __ __ __ 
  A3 __ __ __ __ __ 
  A3 __ __ __ __ __ 
  A2 __ __ __ __ __ 
  A2 __ __ __ __ __ 
  A1 B1 B1 C1 C1 C1 
->put("A3",[1, 3],[0, 3])
  System State: (7) e03: ID already exists
->put("a9",[1, 3],[0, 3])
  System State: (8) e09: Explosive will overlap with another explosive
->suggest([1, 1])
  System State: (9) 
  _____[1,5][2,5][3,5][4,5][5,5]
  _____[1,4][2,4][3,4][4,4][5,4]
  _____[1,3][2,3][3,3][4,3][5,3]
  _____[1,2][2,2][3,2][4,2][5,2]
  _____[1,1][2,1][3,1][4,1][5,1]
  ______________________________
->suggest([2, 2])
  System State: (10) 
  ______________________________
  _____[1,4][2,4][3,4][4,4]_____
  _____[1,3][2,3][3,3][4,3]_____
  _____[1,2][2,2][3,2][4,2]_____
  _____[1,1][2,1][3,1][4,1]_____
  ______________________________
->put("z0",[1, 1],[3, 4])
  System State: (11) 
  A3 __ __ __ __ __ 
  A3 __ __ z0 __ __ 
  A3 __ __ __ __ __ 
  A2 __ __ __ __ __ 
  A2 __ __ __ __ __ 
  A1 B1 B1 C1 C1 C1 
->suggest([2, 2])
  System State: (12) 
  ______________________________
  _____[1,4]__________[4,4]_____
  _____[1,3]__________[4,3]_____
  _____[1,2][2,2][3,2][4,2]_____
  _____[1,1][2,1][3,1][4,1]_____
  ______________________________
->put("B2",[2, 2],[1, 1])
  System State: (13) 
  A3 __ __ __ __ __ 
  A3 __ __ z0 __ __ 
  A3 __ __ __ __ __ 
  A2 B2 B2 __ __ __ 
  A2 B2 B2 __ __ __ 
  A1 B1 B1 C1 C1 C1 
->put("B3",[2, 3],[1, 3])
  System State: (14) 
  A3 B3 B3 __ __ __ 
  A3 B3 B3 z0 __ __ 
  A3 B3 B3 __ __ __ 
  A2 B2 B2 __ __ __ 
  A2 B2 B2 __ __ __ 
  A1 B1 B1 C1 C1 C1 
->put("C2",[3, 2],[3, 1])
  System State: (15) 
  A3 B3 B3 __ __ __ 
  A3 B3 B3 z0 __ __ 
  A3 B3 B3 __ __ __ 
  A2 B2 B2 C2 C2 C2 
  A2 B2 B2 C2 C2 C2 
  A1 B1 B1 C1 C1 C1 
->put("C3",[3, 3],[3, 3])
  System State: (16) e09: Explosive will overlap with another explosive
->remove("B2")
  System State: (17) 
  A3 B3 B3 __ __ __ 
  A3 B3 B3 z0 __ __ 
  A3 B3 B3 __ __ __ 
  A2 __ __ C2 C2 C2 
  A2 __ __ C2 C2 C2 
  A1 B1 B1 C1 C1 C1 
->remove("A1")
  System State: (18) 
  A3 B3 B3 __ __ __ 
  A3 B3 B3 z0 __ __ 
  A3 B3 B3 __ __ __ 
  A2 __ __ C2 C2 C2 
  A2 __ __ C2 C2 C2 
  __ B1 B1 C1 C1 C1 
->remove("C3")
  System State: (19) e10: ID does not exist
->suggest([2, 2])
  System State: (20) 
  ______________________________
  ____________________[4,4]_____
  ____________________[4,3]_____
  ______________________________
  _____[1,1]____________________
  ______________________________
->suggest([1, 1])
  System State: (21) 
  _______________[3,5][4,5][5,5]
  ____________________[4,4][5,4]
  _______________[3,3][4,3][5,3]
  _____[1,2][2,2]_______________
  _____[1,1][2,1]_______________
  [0,0]_________________________
->suggest([3, 3])
  System State: (22) 
  ______________________________
  ______________________________
  ______________________________
  ______________________________
  ______________________________
  ______________________________
